sdk/core/core-client/src/dom.d.ts

```diff
@@ -0,0 +1,4 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom" />
```

Comment:
I copied this from `core-http` because I was getting annoying compile errors, but I think it won't require external users to import that lib since they're not compiling our source and we don't re-export types that are part of `dom`.

It just is supposed to give us typing for things that are from `dom` when we feature detect them (e.g. in serializer.ts when it checks if Blob exists.)

/cc @willmtemple who knows a thing or two because he's seen a thing or two